### DISCUSSION TITLE
p2p storage in git-bug

### DISCUSSION THREAD

**Discussion by ghost** [git-bug/git-bug#888] [Q&A] [answered]
Hi.

Recently I was wondering if it would be possible or feasible... to add p2p storage in git-bug, so I could share bugs with network nodes.  As an example of p2p storage, there are network protocols such as IPFS, Hypercore, Bittorrent. These protocols are often used both for storing and sharing things. 

I would like to know what you all think of the idea... in the long term I'm thinking about adding these network protocols p2p in git-bug, but I need the community's opinion if this is interesting or not.

**Comment by MichaelMure** [comment, marked as answer]
It's doable and would be a fun thing to have. Some design choices (notably the id space) have been made with that sort of thing in mind.

I'd expect the easiest way would be to work at the git level, with one of those p2p git remote implementation, but it might also be possible to implement an alternative Repository in git-bug.

**Reply by ghost** [reply]
Hi MichaelMure, thank you for feedback.
